packages/flutter/lib/src/material/autocomplete.dart

```diff
@@ -156,7 +164,7 @@ class _AutocompleteOptions<T extends Object> extends StatelessWidget {
       child: Material(
         elevation: 4.0,
         child: ConstrainedBox(
-          constraints: BoxConstraints(maxHeight: maxOptionsHeight),
+          constraints: BoxConstraints(maxHeight: maxOptionsHeight, maxWidth: maxOptionsWidth),
```

Comment:
What if instead of maxOptionsHeight and maxOptionsWidth we had a single optionsConstraints parameter that took a whole BoxConstraints?  That would require deprecating maxOptionsHeight but it shouldn't be too hard.